pkg/rke2/machine_filters_test.go

```diff
@@ -0,0 +1,101 @@
+package rke2
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/collections"
+
+	bootstrapv1 "github.com/rancher-sandbox/cluster-api-provider-rke2/bootstrap/api/v1alpha1"
+	controlplanev1 "github.com/rancher-sandbox/cluster-api-provider-rke2/controlplane/api/v1alpha1"
+)
+
+var rcp = controlplanev1.RKE2ControlPlane{
```

Comment:
Might be worth wrapping this in a function that returns a pointer to a new instance every time.

Further down the line when more tests are added it become problematic.

Reply:
Tracking this in issue #80